pkg/kubelet/config/common_test.go

```diff
@@ -87,6 +90,17 @@ func TestDecodeSinglePod(t *testing.T) {
 		t.Errorf("expected:\n%#v\ngot:\n%#v\n%s", pod, podOut, string(json))
 	}
 
+	parsed, podOut, err = tryDecodeSinglePod(json, errDefault)
```

Comment:
shouldn't this be a table test instead of injecting it here?
`tryDecodeSinglePod` is called later too, inside the for loop

Reply:
@aojea   I use `go test -coverprofile=coverage.out` and `go tool cover -html=coverage.out` and notice that the following code has not be covered

```
        if len(pod.UID) == 0 {
                hasher := md5.New()
                hash.DeepHashObject(hasher, pod)
                // DeepHashObject resets the hash, so we should write the pod source
                // information AFTER it.
                if isFile {
                        fmt.Fprintf(hasher, "host:%s", nodeName)
                        fmt.Fprintf(hasher, "file:%s", source)
                } else {
                        fmt.Fprintf(hasher, "url:%s", source)
                }
                pod.UID = types.UID(hex.EncodeToString(hasher.Sum(nil)[0:]))
                klog.V(5).InfoS("Generated UID", "pod", klog.KObj(pod), "podUID", pod.UID, "source", source)
        }
```

So I write a unit test for `tryDecodeSinglePod` to cover this ifstatement

Reply:
Yeah, but it is still the same test with one additional call, and it seems we are mixing different scenarios in the same test.

Is not better if we have a table test with the different scenarios instead of mixing them?

Reply:
Thanks for your reply, I will try and close this PR